### PR TITLE
BSA-18 | Logic for turd frequency

### PR DESCRIPTION
**Pull request by danmcgill6**
- Extracted turd logic

### CODE REVIEW

File: engine/src/systems/bird.js [hunk 1]
```diff
@@ -1,112 +1,54 @@
...
-      delete entities[t]   
-    }
-  })
+  increaseLevel(seconds, level, levelTime, totalBirds, birdInterval);
+  generateNewBird(physics, seconds, entities, birdInterval);
+  deleteOutOfBoundsBirds(physics.world, entities);
+  checkIfTurdHasHitPlayer(entities, character, physics);
 
+  let birds = getBirds(entities);
+  // If birds are on the screen
```
Comment (ohmlnz) [inline engine/src/systems/bird.js:32]: That comment is not totally correct. We're simply checking if there are any birds in the current scene. They can be hidden, especially at the beginning of a level. As a matter of fact, we only start to move them around on line 37 with `updateBirdPosition`

File: engine/src/systems/character.js [hunk 2]
```diff
@@ -30,7 +30,7 @@ export default (entities, { touches, dispatch }) => {
     // acceleration may be based upon power-ups or set statically
     character.acceleration = 1 //+= 0.03;
     let { position, velocity } = character.body;
-    Matter.Body.setVelocity(character.body, { x: 3.5 * character.acceleration, y: velocity.y });
+    Matter.Body.setVelocity(character.body, { x: 5.5 * character.acceleration, y: velocity.y });
```
Comment (ohmlnz) [inline engine/src/systems/character.js:33]: 🏃‍♂️💨

**Review by ohmlnz — CHANGES_REQUESTED** [review]

File: engine/src/systems/helpers/bird.js [hunk 3]
```diff
@@ -41,9 +49,41 @@ function deleteBirds(currentWorld, entities) {
...
+  if(seconds >= entities.interval) {
+    let birdName = '__bird__' + uuidv4();
+    let newBird = addBird(physics.world, width, birdName);
+    newBird.poopInterval = seconds + Math.random() * 10;
+    entities.birds++;
+    if (!newBird.isGoingLeft)
+      newBird.direction.horizontal = 'left';
+    // Generate turds for new birds
+    let turd = Turd(physics.world, { x: -50, y: newBird.body.position.y });
+    newBird.turds = new Array(randomized(5, 1)).fill(turd);
```
[the remark this answers is not in the document]
Comment (danmcgill6) [reply engine/src/systems/helpers/bird.js:76]: Ahhhhhhhhhhhhhh nice catch, makes sense.

File: engine/src/systems/bird.js (hunk 1 above)
Comment (danmcgill6) [reply engine/src/systems/bird.js:32]: @ohmlnz Agreed, I still cant push to the branch thoe to change it

File: engine/src/systems/character.js (hunk 2 above)
Comment (danmcgill6) [reply engine/src/systems/character.js:33]: hehehehhehehehe

**Review by ohmlnz — APPROVED** [review]

**Pull request merged**